cypress/e2e/claim-portal-failure.cy.ts

```diff
@@ -0,0 +1,193 @@
+
+      cy.get("button[id='make-claim']").should("be.visible").click();
+      cy.get("#invalidator").should("not.be.visible");
+      cy.get("#claim-loader").should("not.be.visible");
+      cy.get("#view-claim").should("not.be.visible").and("include.text", "View Claim");
+
+      cy.get("body").should("contain.text", "This reward is not for you");
+    });
+  });
+
+  describe("Failed transactions", () => {
+    const provider = new JsonRpcProvider("http://127.0.0.1:8545");
+    const signer = provider.getSigner();
+
+    beforeEach(() => {
+      cy.clearAllCookies();
+      cy.clearAllLocalStorage();
+      cy.clearAllSessionStorage();
+
+      setupIntercepts();
+      stubEthereum(beneficiary, signer);
+
+      cy.visit(`/${claimUrl}`);
+      cy.wait(2000);
+    });
+
+    it("should handle feedback for a failed wallet provider transaction", () => {
+      cy.get("#additionalDetails", { timeout: 15000 }).should("be.visible").invoke("click");
+
+      cy.get("button[id='make-claim']").should("be.visible").click();
+      cy.get("#claim-loader").should("be.visible");
+      cy.get("#invalidator").should("not.be.visible");
+      // cy.get("#claim-loader").should("not.be.visible"); // gets stuck here
+    });
+  });
+});
+
+function setupIntercepts() {
+  cy.intercept("POST", "*", (req) => {
+    // return a 404 for rpc optimization meaning no successful RPC
+    // to return our balanceOf and allowance calls
+    if (req.body.method == "eth_getBlockByNumber") {
+      req.reply({
+        statusCode: 404,
+        body: {
+          jsonrpc: "2.0",
+          error: {
+            code: -32601,
+            message: "Method not found",
+          },
+          id: 1,
+        },
+      });
+    }
+
+    if (req.body.method == "eth_sendTransaction") {
+      req.reply({
+        statusCode: 404,
+        body: {
+          jsonrpc: "2.0",
+          id: 44,
+          result: "0x",
+        },
+      });
+    }
+
+    if (req.body.method == "eth_call") {
+      const selector = req.body.params[0].data.slice(0, 10);
+
+      // balanceOf
+      if (selector == "0x70a08231") {
+        req.reply({
+          statusCode: 200,
+          body: {
+            jsonrpc: "2.0",
+            id: 45,
+            result: "0x00000000000000000000000000000000000000000000478cf7610f95b9e70000",
+          },
+        });
+      } else if (selector == "0xdd62ed3e") {
+        //  allowance
+
+        req.reply({
+          statusCode: 200,
+          body: {
+            jsonrpc: "2.0",
+            id: 46,
+            result: "0x0000000000000000000000000000000000c097ce7bc906e58377f59a8306ffff",
+          },
+        });
+      } else if (selector == "0x95d89b41" || selector == "0x313ce567" || selector == "0x4fe02b44") {
+        // decimals and symbol
+        // get names?
+        // nonceBitmap
+      } else if (selector == "0xcbf8b66c") {
+        // permit
+        // req.destroy();
```

Comment:
What is all of this section for?

Reply:
I was covering all tracks and realised they were not immediately needed but could be used to expand the tests possibly, so I kept it since the legwork was already done.

I'll remove them

Reply:
@gentlementlegen resolved